fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/metadata/RecordMetaDataBuilderTest.java

```diff
@@ -339,4 +340,33 @@ public void noPrimaryKey() {
                 RecordMetaData.newBuilder().setRecords(TestRecordsNoPrimaryKeyProto.getDescriptor()).getRecordMetaData());
         assertEquals("Record type MyNoPrimaryKeyRecord must have a primary key", e.getMessage());
     }
+
+    @Test
+    public void updateRecords() {
+        RecordMetaDataBuilder builder = RecordMetaData.newBuilder().setRecords(TestRecords1Proto.getDescriptor());
+        final int prevVersion = builder.getVersion();
+        assertNotNull(builder.getRecordType("MySimpleRecord"));
+        assertSame(builder.getRecordType("MySimpleRecord").getDescriptor().getFile(), TestRecords1Proto.getDescriptor());
+        assertNotNull(builder.getRecordType("MyOtherRecord"));
+        builder.updateRecords(TestRecords1EvolvedProto.getDescriptor(), true);
+        RecordMetaData recordMetaData = builder.build(true);
+        assertNotNull(recordMetaData.getRecordType("MySimpleRecord"));
+        assertSame(builder.getRecordType("MySimpleRecord").getDescriptor().getFile(), TestRecords1EvolvedProto.getDescriptor());
+        assertNotNull(recordMetaData.getRecordType("MyOtherRecord"));
+        assertSame(builder.getRecordType("MyOtherRecord").getDescriptor().getFile(), TestRecords1EvolvedProto.getDescriptor());
+        assertNotNull(recordMetaData.getRecordType("AnotherRecord"));
+        assertSame(builder.getRecordType("AnotherRecord").getDescriptor().getFile(), TestRecords1EvolvedProto.getDescriptor());
+        assertSame(builder.getRecordType("AnotherRecord").getSinceVersion(), builder.getVersion());
```

Comment:
`assertEquals` rather than `assertSame` here, I think (for the latter assertion).

Reply:
I think that's a reason to *not* to use `assertSame`. That assertion verifies that things are reference-equal rather than logically equal, and I think we only care about logical equality for this value.

If you cast `getSinceVersion` to an `int`, I think you can use `assertEquals`.

Reply:
sure, done.